src/commands/command.ts

```diff
@@ -0,0 +1,53 @@
+import { Message } from 'discord.js';
+import { PermissionsLevel } from '@/services/permissions.service';
+
+export type TriggerCriteria = string | RegExp | ((message: Message, keyword: string, args: string[]) => string | null);
```

Comment:
I wasn't a huge fan of this solution, but it's not a big deal because I simply rely on whether or not the value is truthy/falsy (i.e. a string counts as true, and `null` counts as false), or using the null coalescing operator `??` as needed.

The reason for this is that it's not just returning whether the criteria is met, but what `text` (in theory from the input string) actually passed this trigger. Or, if we wanted to modify that slightly to include some inline tags (i.e. for NLP), that could all be done in this stage.